Start Storybook and react components additions

dalecreativ opened the pull request:
## Storybook and react additions

What we've done:
- Added react and react-dom
- Added Storybook react, for react component development
- Downgraded PostCSS from v8 to v7. See NOTES
- Updated `babel.config.js` for react builds
- Updated the `component-template` package to help start new react components.
- Added a `component-blockquote`, includes a react component, styles and jest tests.


To test:

Start the storybook app
```
npm ci
npm run storybook
```
confirm working with stories at `localhost:6006`


Run a build
```
npm run build
```
Should build all packages without error.

## NOTES:
Storybook react 6.1 > 6.2 has had known issues around its dependencies.

This has caused issues with builds for storybook when using packages like `postcss v8` in your own projects that use storybook.  As a temporary measure, we've downgraded the postcss version to 7, until an official patch has been released.
Issues and discussions from storybook:
- https://github.com/storybookjs/storybook/issues/13491
- https://github.com/storybookjs/storybook/issues/12668

UPDATE: Storybook are currently on beta release of 6.2.  With some updates and a additional addon `@storybook/addon-postcss`.  We're now able to build without error.
I've set up a branch to help track beta to production release here : https://github.com/coopdigital/coop-frontend/tree/storybook-beta-with-postcss8

mchadwickweb commented:
@dalecreativ everything is looking great. I think as a first pass for getting React into here this is great.

We have already spoken about the use of React Testing Library over Enzyme for integration level testing.

I think some next steps would be:

1. Test the E2E flow of development, test, publish, consume for the component.
2. Look into how we can ring fence our build configuration from Storybook and remove the current coupling to it.

dalecreativ commented:
@mchadwickweb I've updated the dependencies for react-testing-library and removed deps on enzyme and react-test-renderer.  Made sure to update the tests and snapshots for the components too. 👍
